frontend/src/components/game/PlayerGameView.tsx

```diff
@@ -357,22 +418,40 @@ function PlayerGameView(props: PlayerGameViewProps) {
         // Set the 'submit' submission type to correctly display result.
         // eslint-disable-next-line no-param-reassign
         res.submissionType = SubmissionType.Submit;
-        setSubmission(res);
+        setSubmissions(submissions.concat([res]));
+        setCurrentSubmission(res);
       })
       .catch((err) => {
         setLoading(false);
         setError(err.message);
       });
   };
 
+  const nextProblem = () => {
+    setCurrentProblemIndex((currentProblemIndex + 1) % problems?.length);
+    setCurrentSubmission(getSubmission((currentProblemIndex + 1) % problems?.length, submissions));
+  };
+
+  const previousProblem = () => {
```

Comment:
Note: these functions are updated in #174 to no longer loop between the first and last problem